### PR TITLE
Add ability to assign specific classes to devices

### LINKED ISSUE #874: Stylus not recognized
If I use my Wacom Bamboo Ink stylus it's not recognized as an stylus input, but rather as an input from my touchscreen.
I tested the stylus with the gtk3 touch demo and it is recognized as a different source than the touchscreen.
I don't know if this is an bug or a false configuration.

### LINKED ISSUE #1136: New input spurious dots and strokes when the tablet has hand pad device active
**Affects versions :**
 - OS: Linux Ubuntu 18.04
 - Version of Xournal++: master 290e7a9c, new input system activated

**Describe the bug**
My tablet is an Intuos CTH480 and it has a physical switch to enable the touchpad behavior. When the switch is off, there is no problem at all, the pen works ok, and there are no spurious dots or strokes. 

When I switch the touchpad on, xournalpp uses the touch events to scroll (great!), but sometime it seems it confuses the touch event with the pen, leading to strokes (rarely) or spurious dots (much more common). 

**To Reproduce**
- Activate the touchpad switch on the tablet.
- Draw something and in the middle, scroll (I normally have the hand resting on the tablet, to scroll I hover the hand and the pen, and I use the fourth finger to scroll). 
- Sometimes a dot appears, or a line is drawn, when I approach back the pen to the tablet without touching it. The `key-mon` indicator *seems* to show a middle-mouse click event in this case...  

**Input Logs**
I have two logs with `DDEBUG_INPUT=ON:
- a log when spurious strokes happened (`debug_input_unwanted_strokes.log`)
- a log when spurious points appeared (`debug_input_unwanted_dot.log`)

and one log with `DEBUG_INPUT_GDK_PRINT_EVENTS=ON` when dots appeared (`debug_input_gdk.log`); I was not able for now to reproduce the unwanted strokes with this debug option. 

Zipped logs: 
[logs.zip](https://github.com/xournalpp/xournalpp/files/3168967/logs.zip)

**Expected behavior**

No spurious strokes or dot should appear. 

**Screenshots of Problem**
Three screencasts related to the three logs: 
1. `debug_input_unwanted_strokes.log`: 
unwanted_strokes

2. `debug_input_unwanted_dot.log`
unwanted_dot

3.  `debug_input_gdk.log`
unwanted_dot_gdk

**Additional context**

This is low priority, because shutting off the touchpad is easy and fast.

### LINKED ISSUE #1192: Consider switching default cursor type to large
There's been multiple issues concerning the visibility of the cursor (e.g. see #1191 #1077 #974). I believe that this confusion is frequent enough to warrant changing the default cursor size to the big cursor.

### PR DESCRIPTION
**Pull request by LittleHuba**
Adds functionality to manually assign devices to certain input device classes.

Currently only the UI frontend is implemented. We will need an abstraction layer in the input system for this to work, as events of touchscreens are inherently different from those form a mouse or stylus. Easiest would be to map all touch events to mouse events as we can work with those and only allow one finger at a time anyway...

Fixes #1136 Fixes #874 Fixes #1192

Note:
This PR remodels the settings heavily, as currently they are limited by the number of tabs.
image

### CODE REVIEW

**Comment by LittleHuba** [comment]
Fixes #1136 #874

**Comment by andreasb242** [comment]
@LittleHuba 

Does not Compile on Ubuntu 18.04

[  7%] Building CXX object src/util/CMakeFiles/util.dir/audio/AudioPlayer.cpp.o
[  8%] Building CXX object src/util/CMakeFiles/util.dir/audio/AudioRecorder.cpp.o
In file included from /home/andreas/dev/xournalpp_ulrich/src/control/Control.h:21:0,
                 from /home/andreas/dev/xournalpp_ulrich/src/util/audio/AudioPlayer.cpp:1:
/home/andreas/dev/xournalpp_ulrich/src/gui/MainWindow.h:111:7: error: ‘void MainWindow::setTouchscreenScrollingForDeviceMapping()’ cannot be overloaded
  void setTouchscreenScrollingForDeviceMapping();
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/andreas/dev/xournalpp_ulrich/src/gui/MainWindow.h:105:7: error: with ‘void MainWindow::setTouchscreenScrollingForDeviceMapping()’
  void setTouchscreenScrollingForDeviceMapping();
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/util/CMakeFiles/util.dir/build.make:470: recipe for target 'src/util/CMakeFiles/util.dir/audio/AudioPlayer.cpp.o' failed
make[2]: *** [src/util/CMakeFiles/util.dir/audio/AudioPlayer.cpp.o] Error 1
make[2]: *** Auf noch nicht beendete Prozesse wird gewartet …
CMakeFiles/Makefile2:426: recipe for target 'src/util/CMakeFiles/util.dir/all' failed
make[1]: *** [src/util/CMakeFiles/util.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2

**Comment by LittleHuba** [comment]
Just fixed this. Was a merge conflict.

**Comment by LittleHuba** [comment]
Implementation is now finished. There are still some issues with Windows but fixes will follow in other PRs. The main functionality is in now.

Drawing with touchscreen will also be fixed in another PR. This will deprecate the GTK workaround if possible.

File: src/gui/inputdevices/MouseInputHandler.cpp [hunk 1]
```diff
@@ -54,9 +54,16 @@ bool MouseInputHandler::handleImpl(InputEvent* event)
 	 * Trigger motion actions
 	 */
 	// Trigger motion action when pen/mouse is pressed and moved
-	if (this->deviceClassPressed && event->type == MOTION_EVENT) //mouse or pen moved
+	if (event->type == MOTION_EVENT) //mouse or pen moved
 	{
-		this->actionMotion(event);
+		if (this->deviceClassPressed)
+		{
```
Comment (JJones780) [inline src/gui/inputdevices/MouseInputHandler.cpp]: I tried :
```
XournalppCursor* cursor = xournal->view->getCursor();
			cursor->updateCursor();
```
here to get teh custom cursor during mouse down. What do you think?

Comment (LittleHuba) [reply src/gui/inputdevices/MouseInputHandler.cpp]: I did not really get what you mean. Could you elaborate?

**Pull request merged**